src/wallet/crypter.cpp

```diff
@@ -135,13 +136,31 @@ static bool DecryptKey(const CKeyingMaterial& vMasterKey, const std::vector<unsi
     return key.VerifyPubKey(vchPubKey);
 }
 
+static bool DecryptSpendingKey(const CKeyingMaterial& vMasterKey, const std::vector<unsigned char>& vchCryptedSecret, const libzcash::PaymentAddress& address, libzcash::SpendingKey& key)
+{
+    CKeyingMaterial vchSecret;
+    if(!DecryptSecret(vMasterKey, vchCryptedSecret, address.GetHash(), vchSecret))
+        return false;
+
+    if (vchSecret.size() != 32)
+        return false;
+
+    // TODO does this undo the benefits of using CKeyingMaterial?
```

Comment:
https://github.com/zcash/zcash/blob/zc.v0.11.2.latest/src/wallet/crypter.h#L67 :
``typedef std::vector<unsigned char, secure_allocator<unsigned char> > CKeyingMaterial;``

So it should be possible to do:
``CDataStream ss(vchSecret, SER_NETWORK, PROTOCOL_VERSION);``

(The ``SpendingKey`` class should itself be using a secure allocator but that's a separate issue.)